ArduPlane/Plane.h

```diff
@@ -454,6 +459,11 @@ class Plane : public AP_Vehicle {
         float throttle_lim_min;
         uint32_t throttle_max_timer_ms;
         uint32_t level_off_start_time_ms;
+        // Good candidate for keeping the initial time for TKOFF_THR_MAX_T.
+#if MODE_AUTOLAND_ENABLED
+        int32_t takeoff_initial_direction; //deg
+        bool takeoff_direction_initialized;
+#endif
```

Comment:
Comment is out of date and int32 is a bit unexpected for direction, why not float?
```suggestion
#if MODE_AUTOLAND_ENABLED
        struct {
            float heading; // deg
            bool initialized;
        } initial_direction;
#endif
```
You could also add a little structure to keep the variable names shorter.

Reply:
done